### PR TITLE
Resolve edge cases with `--tmpdir` behavior

### PR DESCRIPTION
**Pull request by Racer159**
## Description

This PR resolves edge cases with `--tmpdir` behavior by always using the specified temp dir in commands and always making a unique sub folder for each zarf run.

## Related Issue

Fixes #N/A

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Other (security config, docs update, etc)

## Checklist before merging

- [X] Test, docs, adr added or updated as needed
- [X] [Contributor Guide Steps](https://github.com/defenseunicorns/zarf/blob/main/CONTRIBUTING.md#developer-workflow) followed

### CODE REVIEW

**Review by cmwylie19 — DISMISSED** [review]
lgtm

File: src/pkg/utils/io.go [hunk 1]
```diff
@@ -59,8 +59,13 @@ type TextTemplate struct {
 }
 
 // MakeTempDir creates a temp directory with the zarf- prefix.
-func MakeTempDir() (string, error) {
-	tmp, err := os.MkdirTemp("", tmpPathPrefix)
+func MakeTempDir(basePath string) (string, error) {
+	if basePath != "" && InvalidPath(basePath) {
```
Comment (Noxsios) [inline src/pkg/utils/io.go:63]:
```suggestion
	if basePath != "" {
```

`utils.CreateDirectory` already performs a `InvalidPath` check

File: src/pkg/utils/io.go [hunk 2]
```diff
@@ -59,8 +59,13 @@ type TextTemplate struct {
...
+		if err := CreateDirectory(basePath, 0700); err != nil {
+			return "", err
+		}
+	}
+	tmp, err := os.MkdirTemp(basePath, tmpPathPrefix)
 	message.Debugf("Creating temp path: '%s'", tmp)
```
Comment (Noxsios) [inline src/pkg/utils/io.go:69]:
```suggestion
	message.Debugf("Created temp directory: %q", tmp)
```

**Review by Noxsios — CHANGES_REQUESTED** [review]

File: src/pkg/utils/io.go (hunk 2 above)
Comment (Racer159) [reply src/pkg/utils/io.go]: Didn't exactly do this but normalized the verbiage across things - noticed we were double printing the path in some cases as well.

**Review by Noxsios — APPROVED** [review]

**Pull request merged**